pkg/openapi/run.go

```diff
@@ -0,0 +1,453 @@
+		return "", false, fmt.Errorf("failed to parse server URL %s: %w", opInfo.Server+opInfo.Path, err)
+	}
+
+	// Set up the request
+	req, err := http.NewRequest(opInfo.Method, u.String(), nil)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Check for authentication (only if using HTTPS or localhost)
+	if u.Scheme == "https" || u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
+		if len(opInfo.SecurityInfos) > 0 {
+			if err := HandleAuths(req, envMap, opInfo.SecurityInfos); err != nil {
+				return "", false, fmt.Errorf("error setting up authentication: %w", err)
+			}
+		}
+
+		// If there is a bearer token set for the whole server, and no Authorization header has been defined, use it.
+		if token, ok := envMap["GPTSCRIPT_"+env.ToEnvLike(u.Hostname())+"_BEARER_TOKEN"]; ok {
+			if req.Header.Get("Authorization") == "" {
+				req.Header.Set("Authorization", "Bearer "+token)
+			}
+		}
+	} else {
+		fmt.Fprintf(os.Stderr, "no auth")
+	}
+
+	// Handle query parameters
+	req.URL.RawQuery = HandleQueryParameters(req.URL.Query(), opInfo.QueryParams, args).Encode()
+
+	// Handle header and cookie parameters
+	HandleHeaderParameters(req, opInfo.HeaderParams, args)
+	HandleCookieParameters(req, opInfo.CookieParams, args)
+
+	// Handle request body
+	if opInfo.BodyContentMIME != "" {
+		res := gjson.Get(args, "requestBodyContent")
+		var body bytes.Buffer
+		switch opInfo.BodyContentMIME {
+		case "application/json":
+			var reqBody interface{}
+
+			reqBody = struct{}{}
+			if res.Exists() {
+				reqBody = res.Value()
+			}
+			if err := json.NewEncoder(&body).Encode(reqBody); err != nil {
+				return "", false, fmt.Errorf("failed to encode JSON: %w", err)
+			}
+			req.Header.Set("Content-Type", "application/json")
+
+		case "text/plain":
+			reqBody := ""
+			if res.Exists() {
+				reqBody = res.String()
+			}
+			body.WriteString(reqBody)
+
+			req.Header.Set("Content-Type", "text/plain")
+
+		case "multipart/form-data":
+			multiPartWriter := multipart.NewWriter(&body)
+			req.Header.Set("Content-Type", multiPartWriter.FormDataContentType())
+			if res.Exists() && res.IsObject() {
+				for k, v := range res.Map() {
+					if err := multiPartWriter.WriteField(k, v.String()); err != nil {
+						return "", false, fmt.Errorf("failed to write multipart field: %w", err)
+					}
+				}
+			} else {
+				return "", false, fmt.Errorf("multipart/form-data requires an object as the requestBodyContent")
+			}
+			if err := multiPartWriter.Close(); err != nil {
+				return "", false, fmt.Errorf("failed to close multipart writer: %w", err)
+			}
+
+		default:
+			return "", false, fmt.Errorf("unsupported MIME type: %s", opInfo.BodyContentMIME)
+		}
+		req.Body = io.NopCloser(&body)
+	}
+
+	// Make the request
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to make request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	result, err := io.ReadAll(resp.Body)
```

Comment:
To be on the safe side, this could use a nil check to ensure that the Body exists